spec.md

```diff
@@ -1459,6 +1477,21 @@ The `else`{.firrtl} branch may also be added to the single line:
 when c : a <= b else : e <= f
 ```
 
+### Match Statements
+
+Match statements are used to discriminate the active variant of a enumeration
+typed expression.  A match statement must exhaustively test every variant of an
+enumeration. An optional binder may be specified to extract the data of the
+variant.
+
+``` firrtl
+match x:
+  #some(v):
```

Comment:
final bikesheding round:
If identifiers aren't prefixed in the type, no reason to prefix them here.